### PR TITLE
[WIP] Add memory measurements to interpreted PyROOT

### PR DESCRIPTION
**Pull request by HLilit**
Improve memory measurements for ROOT interpreter benchmark

### CODE REVIEW

File: lib/MemoryMeasurement.cxx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#include <iostream>
+#include <cstdlib>
+#include <string>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <unistd.h>
+#include <chrono>
+#include <stdlib.h>
+#include "rootbench/RBConfig.h"
```
Comment (vgvassilev) [inline lib/MemoryMeasurement.cxx]: Includes should be in reverse order, that is, less generic to more generic.

File: lib/MemoryMeasurement.cxx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include "rootbench/MemoryMeasurement.h"
+
+std::string getFileName(std::string path) {
+	std::string base_fn = path.substr(path.find_last_of("/\\") + 1);
+	return base_fn.substr(0, base_fn.find_last_of('.'));
+}
+
+std::string getMemoryStats(const std::string& dir, const std::string& filename, std::string& rootInvocation, const std::string& rootexe, const std::string& endInvoc) {
+		std::string perffile="";
+		if (access("/usr/bin/time", X_OK) == 0) {
```
Comment (vgvassilev) [inline lib/MemoryMeasurement.cxx]: We should not hardcode this path `/usr/bin/time`, we should use a more standard way to resolve a system binary.

File: lib/MemoryMeasurement.cxx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			std::string rootsys = RB::GetRootSys();
+			std::string thisroot = rootsys + "/bin/thisroot.sh";
+			perffile = getFileName(filename) + "_perfile.txt";
+			if (!filename.empty()) {
+				std::string fullpath = rootsys + "/" + dir + filename;
+				// FIXME: no source in /usr/dash
+		    // We are writing /usr/bin/time -v output in file to get maximum resident memory for the benchmark
+				rootInvocation = "/usr/bin/time -v -o \"" + perffile + rootexe + fullpath + endInvoc;
+				std::cout<<"helllloooooooooooooooooooo     "<< rootInvocation <<"      hellllllloooooooooooooo"<<std::endl;
```
Comment (vgvassilev) [inline lib/MemoryMeasurement.cxx]: Looks like debug printouts.

File: lib/MemoryMeasurement.cxx (hunk 1 above)
Comment (HLilit) [reply lib/MemoryMeasurement.cxx]: Thanks! I will change the order.

File: lib/MemoryMeasurement.cxx (hunk 3 above)
Comment (HLilit) [reply lib/MemoryMeasurement.cxx]: Yes sorry, forgot to remove them.

File: lib/MemoryMeasurement.cxx (hunk 2 above)
Comment (HLilit) [reply lib/MemoryMeasurement.cxx]: alright! I will try to change it.

**Comment by oshadura** [comment]
Closing in favor of https://github.com/root-project/rootbench/pull/190

**Pull request closed without merging**